test/model/src/include/model/table.h

```diff
@@ -0,0 +1,157 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef MODEL_TABLE_H
+#define MODEL_TABLE_H
+
+#include <atomic>
+#include <map>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "model/core.h"
+#include "model/verify.h"
+#include "wiredtiger.h"
+
+namespace model {
+
+/*
+ * kv_table --
+ *     A database table with key-value pairs.
+ */
+class kv_table {
+
+public:
+    /*
+     * kv_table::kv_table --
+     *     Create a new instance.
+     */
+    inline kv_table(const char *name) : _name(name) {}
+
+    /*
+     * kv_table::name --
+     *     Get the name of the table.
+     */
+    inline const char *
+    name() const noexcept
+    {
+        return (_name.c_str());
+    }
+
+    /*
+     * kv_table::contains_any --
+     *     Check whether the table contains the given key-value pair. If there are multiple values
+     *     associated with the given timestamp, return true if any of them match.
+     */
+    bool contains_any(
+      const data_value &key, const data_value &value, uint64_t timestamp = UINT64_MAX);
```

Comment:
The meaning of the default timestamp param values is unclear to me as I am not familiar with the existing system.
I'd like to suggest something like the following as I think it helps with readability in general:
```
#include <cassert>
#include <cstdint>
#include <limits>

using timestamp_t = std::uint64_t;
constexpr auto TimestampNone = std::numeric_limits<timestamp_t>::max();
constexpr auto TimestampDefault = std::numeric_limits<timestamp_t>::min();

static_assert(TimestampNone == TimestampNone);
static_assert(TimestampDefault == TimestampDefault);
static_assert(TimestampNone != TimestampDefault);
static_assert(TimestampDefault == timestamp_t{0ULL});
static_assert(TimestampDefault == timestamp_t{});
static_assert(TimestampDefault == 0);
```

Reply:
That's a good criticism. I defined `timestamp_t` using `uint64_t` to match WiredTiger's `wt_timestamp_t`. I also re-defined `WT_TS_NONE` and `WT_TS_MAX`, plus the appropriate comments explaining them – I think this would make the connection clearer to the existing constants `WT_TS_NONE` and `WT_TS_MAX` that we use in the main code (but are not defined here, as we are not importing the internal header files).